core/trashcan.js

```diff
@@ -81,7 +81,7 @@ Blockly.Trashcan.prototype.MARGIN_SIDE_ = 20;
  * @type {number}
  * @private
  */
-Blockly.Trashcan.prototype.MARGIN_HOTSPOT_ = 25;
+Blockly.Trashcan.prototype.MARGIN_HOTSPOT_ = 10;
```

Comment:
Is this a UI change?  We increased the hotspot as a result of seeing frustrated children accumulate a bunch of badly-aimed blocks around the trash.

Otherwise, this PR looks great!  Thanks for not scolding me about BIG_NUM; not my proudest code.